pages/devfiles/[id]/index.tsx

```diff
@@ -15,79 +19,74 @@ interface Path {
-  devfileText,
+  devfileYAML,
   devfileJSON,
+  errors
 }: InferGetStaticPropsType<typeof getStaticProps>) => (
   <div style={{ alignContent: 'center', minHeight: '100vh' }}>
+    <ErrorBanner errors={errors} />
     {devfile.type === 'stack' ? (
       <div>
-        <DevPageHeader
-          devfileMetadata={devfileJSON.metadata}
-          devfile={devfile}
-        />
-        <DevPageProjects starterProjects={devfileJSON.starterProjects} />
-        <DevPageYAML devYAML={devfileText} />
+        <DevfilePageHeader devfileMetadata={devfileJSON.metadata} devfile={devfile} />
+        <DevfilePageProjects starterProjects={devfileJSON.starterProjects} />
+        <DevfilePageYAML devfileYAML={devfileYAML} />
       </div>
     ) : (
-      <DevPageHeader devfile={devfile} />
+      <DevfilePageHeader devfile={devfile} />
     )}
   </div>
 );
 
 export const getStaticProps: GetStaticProps = async (context) => {
-  const indexResponse: Response = await fetch(
-    'https://registry.devfile.io/index/all?icon=base64'
-  );
-  const devfiles: Devfile[] = (await indexResponse.json()) as Devfile[];
-  const devfile: Devfile = devfiles.find(
-    (devfile: Devfile) => devfile.name === context.params?.id
-  )!;
+  const [devfiles, devfileErrors]: GetMetadataOfDevfiles = await getMetadataOfDevfiles();
+
+  const devfile: Devfile = devfiles.find((devfile: Devfile) => {
+    const id = context.params?.id as string;
+    const [source, name] = id.split('+');
+    return devfile.sourceRepo === source && devfile.name === name;
+  })!;
 
-  let devfileYAMLResponse: Response;
-  let devfileText: string | null = null;
-  let devfileJSON = null;
+  const [devfileYAML, devfileJSON, yamlErrors]: GetDevfileYAML = await getDevfileYAML(devfile);
 
-  if (devfile.type === 'stack') {
-    devfileYAMLResponse = await fetch(
-      'https://registry.devfile.io/devfiles/' + devfile.name,
-      {
-        headers: { 'Accept-Type': 'text/plain' },
-      }
-    );
-    devfileText = await devfileYAMLResponse.text();
+  const errors = [...devfileErrors, ...yamlErrors];
 
-    // convert yaml text to json
-    const yaml = require('js-yaml');
-    devfileJSON = yaml.load(devfileText);
+  let revalidateSeconds = 15;
+  if (firstTimeRegeneration) {
+    revalidateSeconds = 3;
+    firstTimeRegeneration = false;
   }
 
   return {
     props: {
       devfile,
-      devfileText,
+      devfileYAML,
       devfileJSON,
+      errors
     },
-    revalidate: 21600, // Regenerate page every 6 hours
+    // Next.js will attempt to re-generate the page:
+    // - When a request comes in
+    // - At most once every 15 seconds
+    revalidate: revalidateSeconds
   };
 };
 
 export const getStaticPaths: GetStaticPaths = async () => {
-  const res: Response = await fetch(
-    'https://registry.devfile.io/index/all?icon=base64'
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [devfiles, errors]: GetMetadataOfDevfiles = await getMetadataOfDevfiles();
```

Comment:
errors returned, but nothing is done with with value

Reply:
that is to be expected I cannot do anything with the error on `getStaticPaths`